### ISSUE TITLE
Helm installation of scheduler-estimator fails with cert error

### ISSUE THREAD

**Issue by mszacillo** [karmada-io/karmada#5322]
**What happened**:
Using the most recent version of the karmada helm chart, I attempted to install karmada with the scheduler-estimator component. The schedule-estimator pods all end up in an errored state with the following loglines:

```
I0807 19:42:49.262132       1 scheduler-estimator.go:116] karmada-scheduler-estimator version: version.Info{GitVersion:"v1.11.0-alpha.0-239-g9fa537ea9-dirty", GitCommit:"9fa537ea948a6d2a6e8aa0bfac45147a3f6a727c", GitTreeState:"dirty", BuildDate:"2024-08-05T14:50:48Z", GoVersion:"go1.22.4", Compiler:"gc", Platform:"linux/amd64"}
I0807 19:42:49.264097       1 server.go:152] Starting karmada cluster(spobq2-cluster) accurate scheduler estimator
I0807 19:42:49.465376       1 server.go:168] Listening port: 10352
I0807 19:42:49.465892       1 server.go:173] Shutting down cluster(spobq2-cluster) accurate scheduler estimator
E0807 19:42:49.465910       1 scheduler-estimator.go:138] Estimator server exits unexpectedly: failed to create grpc server: open /etc/karmada/pki/ca.crt: no such file or directory
E0807 19:42:49.465958       1 run.go:74] "command failed" err="failed to create grpc server: open /etc/karmada/pki/ca.crt: no such file or directory"
```

The root cause is that the volume that is mounted on the pod (`karmada-cert` on path `/etc/karmada/pki`) does not contain `ca.crt`. Instead, the associated secret is defined with a `server-ca.crt` and `server-ca.key`.  We fixed this by reapplying a new secret, copying the contents of `karmada-cert` and renaming server-ca.crt to ca.crt. 

**What you expected to happen**:

We expect scheduler-estimator servers to come up successfully after a helm install.

**How to reproduce it (as minimally and precisely as possible)**:

1. Install karmada using the most recent version of the helm chart by running `helm upgrade --install karmada -n karmada-system --create-namespace --dependency-update --cleanup-on-fail --set components={"schedulerEstimator"} .`. The scheduler-estimators will end up in an errored state.

**Environment**:
- Karmada version: v1.11 (unreleased)
- Kubernetes version: v1.29.4

**Comment by mszacillo** [comment]
This also impacts the karmada-scheduler:
```
I0807 21:07:59.214036       1 cache.go:111] Start dialing estimator server(karmada-scheduler-estimator-spobq2-cluster.karmada-system.svc.cluster.local:10352) of cluster(spobq2-cluster).
E0807 21:07:59.214058       1 cache.go:114] Failed to dial cluster(spobq2-cluster): open /etc/karmada/pki/ca.crt: no such file or directory.
I0807 21:07:59.215667       1 cache.go:111] Start dialing estimator server(karmada-scheduler-estimator-sppwq2-new-cluster.karmada-system.svc.cluster.local:10352) of cluster(sppwq2-new-cluster).
E0807 21:07:59.215693       1 cache.go:114] Failed to dial cluster(sppwq2-new-cluster): open /etc/karmada/pki/ca.crt: no such file or directory.
```
Seems like the `server-ca.crt` should be renamed to `ca.crt` in the related secret.

**Comment by mszacillo** [comment]
Easiest fix is to just rename the arguments passed into the scheduler and scheduler-estimator from:
```
--grpc-client-ca-file=/etc/karmada/pki/ca.crt
```
to
```
--grpc-client-ca-file=/etc/karmada/pki/server-ca.crt
```

**Comment by liangyuanpeng** [comment]
same comment: https://github.com/karmada-io/karmada/issues/5345#issuecomment-2285502915

**Comment by chaosi-zju** [comment]
You are right, this is a newly introduced bug by https://github.com/karmada-io/karmada/pull/5097.

https://github.com/karmada-io/karmada/blob/223e54e90d40dcff8fc5133c9ea88b112567aae7/charts/karmada/templates/karmada-scheduler-estimator.yaml#L70-L83

https://github.com/karmada-io/karmada/blob/223e54e90d40dcff8fc5133c9ea88b112567aae7/charts/karmada/templates/_helpers.tpl#L352-L357

from above two code, we learn that path `/etc/karmada/pki` is mount from a secret named `{{ $name }}-cert`, while secret `{{ $name }}-cert` looks that following:

https://github.com/karmada-io/karmada/blob/223e54e90d40dcff8fc5133c9ea88b112567aae7/charts/karmada/templates/pre-install-job.yaml#L249-L269

the ca file name should be `server-ca.crt`, rather than `ca.crt`

**Comment by chaosi-zju** [comment]
I will create a umbrella issue and corresponding PR to unified fix #5322、#5345、#5207

cc @liangyuanpeng
